src/traces/sankey/base_plot.js

```diff
@@ -34,3 +40,94 @@ exports.clean = function(newFullData, newFullLayout, oldFullData, oldFullLayout)
         oldFullLayout._paperdiv.selectAll('.sankey').remove();
     }
 };
+
+exports.updateFx = function(gd) {
+    for(var i = 0; i < gd._fullData.length; i++) {
+        subplotUpdateFx(gd, i);
+    }
+};
+
+var dragOptions = [];
+function subplotUpdateFx(gd, index) {
+    var i = index;
+    var fullData = gd._fullData[i];
+    var fullLayout = gd._fullLayout;
+
+    var dragMode = fullLayout.dragmode;
+    var cursor = fullLayout.dragmode === 'pan' ? 'move' : 'crosshair';
+    var bgRect = fullData._bgRect;
+
+    setCursor(bgRect, cursor);
+
+    var xaxis = {
+        _id: 'x',
+        c2p: function(v) { return v; },
+        _offset: fullData._sankey.translateX,
+        _length: fullData._sankey.width
+    };
+    var yaxis = {
+        _id: 'y',
+        c2p: function(v) { return v; },
+        _offset: fullData._sankey.translateY,
+        _length: fullData._sankey.height
+    };
+
+    // Note: dragOptions is needed to be declared for all dragmodes because
+    // it's the object that holds persistent selection state.
+    dragOptions[i] = {
+        gd: gd,
+        element: bgRect.node(),
+        plotinfo: {
+            id: i,
+            xaxis: xaxis,
+            yaxis: yaxis,
+            fillRangeItems: Lib.noop
+        },
+        subplot: i,
+        // create mock x/y axes for hover routine
+        xaxes: [xaxis],
+        yaxes: [yaxis],
+        doneFnCompleted: function(selection) {
+            var newGroups;
+            var oldGroups = gd._fullData[i].node.groups.slice();
+            var newGroup = [];
+
+            function findNode(pt) {
+                return gd._fullData[i]._sankey.graph.nodes.find(function(n) {
```

Comment:
What's `find` here? Is `nodes` a plain JS array? Is so, we can't use this:

![image](https://user-images.githubusercontent.com/6675409/55415353-dca86b80-553a-11e9-8c7c-8b31721b8993.png)

Reply:
Thank you @etpinard for pointing that out. This `find` has no place in plotly.js. It is replaced by a for loop in 20d8c92d8454cc804e71b84a90469df807d14898